### ISSUE TITLE
NSTimer with blocks doesn’t work (but with @selector works)

### ISSUE THREAD

**Issue by plaurent** [gnustep/libs-base#94]
Hi,

When I use NSTimer with a @selector it works fine, but when I use it with a block it doesn’t work:

```objectivec
  self.nstimer1 = [NSTimer scheduledTimerWithTimeInterval:1.0
                                                  repeats:YES
                                                   block:^(NSTimer* t) {
                                                           NSLog(@"NSTimer1");
                                                         }];
  self.nstimer2 = [NSTimer scheduledTimerWithTimeInterval:1.0
                                                   target:self
                                                 selector:@selector(timer2callback:)
                                                 userInfo:nil
                                                  repeats:YES];

    NSLog(@"NSTimer1 is %@", self.nstimer1);
    NSLog(@"NSTimer2 is %@", self.nstimer2);
}

-(void)timer2callback:(NSTimer*)timer {
    NSLog(@"NSTimer2");
}
```
Note that the block that should be called by NSTimer1 is not called (and, in fact, is printed out as (null)).

```
patryk@pinebook:~/Desktop/gnustep-code/gnustep-code/TimerBugDemo$ openapp ./TimerBugDemo.app
[...]
2020-01-18 01:24:07.250 TimerBugDemo[22133:22133] NSTimer1 is <NSTimer: 0xa254c8> at 2020-01-18 01:24:08 +0000 invokes (null)
2020-01-18 01:24:07.255 TimerBugDemo[22133:22133] NSTimer2 is <NSTimer: 0xa25538> at 2020-01-18 01:24:08 +0000 sends timer2callback: to (<AppDelegate: 0xaf86f8>Number: 4 Title: Window)
2020-01-18 01:24:08.252 TimerBugDemo[22133:22133] NSTimer2
2020-01-18 01:24:09.251 TimerBugDemo[22133:22133] NSTimer2
2020-01-18 01:24:10.251 TimerBugDemo[22133:22133] NSTimer2
2020-01-18 01:24:11.251 TimerBugDemo[22133:22133] NSTimer2
2020-01-18 01:24:12.252 TimerBugDemo[22133:22133] NSTimer2
2020-01-18 01:24:13.251 TimerBugDemo[22133:22133] NSTimer2
```

**Comment by plaurent** [comment]
I have buildable code illustrating this issue at https://github.com/plaurent/TimerBugDemo

**Comment by gcasa** [comment]
Are you compiling this with clang or gcc?

**Comment by plaurent** [comment]
I am compiling this with clang.

**Comment by plaurent** [comment]
@gcasa This is fixed by pull request #95, it was not a clang specific issue, just some bugs in the implementation.

**Comment by gcasa** [comment]
As this is fixed by PR#95, I am closing this issue.
